Source/JavaScriptCore/runtime/ScriptExecutable.cpp

```diff
@@ -304,11 +304,7 @@ CodeBlock* ScriptExecutable::newCodeBlockFor(CodeSpecializationKind kind, JSFunc
         executable->m_unlinkedExecutable->lexicalScopeFeatures(),
         executable->m_unlinkedExecutable->hasCapturedVariables(),
         lastLine(), endColumn()); 
-    // FIXME: We should remove this and add ASSERT(unlinkedCodeBlock).
-    if (!unlinkedCodeBlock) {
-        throwException(globalObject, throwScope, error.toErrorObject(globalObject, executable->source()));
-        return nullptr;
-    }
```

Comment:
Is this correct? What happens when `error` is set?
`UnlinkedFunctionExecutable::unlinkedCodeBlockFor`'s L246 is returning nullptr when error is set.

Reply:
Yes, we should keep the nullptr check for the possible runtime error. And we should also check the exception which might be throw in `UnlinkedModuleProgramCodeBlock::getUnlinkedCodeBlock`.